### PR TITLE
feat!: apps should not depend on other app configs

### PR DESCRIPTION
**Pull request by delta1**
Description
---
Refactors the config example into files for each specific application + common

How Has This Been Tested?
---
Running each app individually with only the configs required for that app 

BREAKING CHANGE!: Old config files may not work and may need to be manually edited

### CODE REVIEW

**Review by sdbondi — APPROVED** [review]
Very happy about the amount of deleted code 😆  Testing now but LGTM

**Review by sdbondi — DISMISSED** [review]
Tested base node with new config and working. Also seems to work fine with the old configs 

EDIT: the base node I tested with already had peers,  so moving peer seeds to `common` is a breaking change

**Comment by delta1** [comment]
Thanks @sdbondi - yeah nice to DRY up that stuff a little lol

**Review by stringhandler — DISMISSED** [review]

**Comment by delta1** [comment]
Added a batch file to be run before creating windows installer

File: buildtools/windows_inno_installer.iss [hunk 1]
```diff
@@ -111,7 +111,8 @@ Source: "..\applications\tari_merge_mining_proxy\windows\runtime\source_merge_mi
 Source: "..\applications\tari_merge_mining_proxy\windows\runtime\start_tari_merge_mining_proxy.bat"; DestDir: "{app}\runtime"; Flags: ignoreversion
 Source: "..\applications\tari_merge_mining_proxy\windows\runtime\source_xmrig_env.bat"; DestDir: "{app}\runtime"; Flags: ignoreversion
 Source: "..\applications\tari_merge_mining_proxy\windows\runtime\start_xmrig.bat"; DestDir: "{app}\runtime"; Flags: ignoreversion
-Source: "..\common\config\presets\tari_config_example.toml"; DestDir: "{app}\config"; DestName: "config.toml"; Flags: ignoreversion
+; Note: please run generate_config.bat to generate this file first 
+Source: ".\tari_config_example.toml"; DestDir: "{app}\config"; DestName: "config.toml"; Flags: ignoreversion
```
Comment (delta1) [inline buildtools/windows_inno_installer.iss:117]: @hansieodendaal @Cifko FYI need to run `buildtools/generate_config.bat` before creating the windows installer

File: buildtools/generate_config.bat [hunk 2]
```diff
@@ -0,0 +1 @@
+type ..\common\config\presets\*.toml >.\tari_config_example.toml
```
Comment (delta1) [inline buildtools/generate_config.bat:1]: @hansieodendaal @Cifko FYI

File: buildtools/windows_inno_installer.iss [hunk 3]
```diff
@@ -111,7 +111,8 @@ Source: "..\applications\tari_merge_mining_proxy\windows\runtime\source_merge_mi
 Source: "..\applications\tari_merge_mining_proxy\windows\runtime\start_tari_merge_mining_proxy.bat"; DestDir: "{app}\runtime"; Flags: ignoreversion
 Source: "..\applications\tari_merge_mining_proxy\windows\runtime\source_xmrig_env.bat"; DestDir: "{app}\runtime"; Flags: ignoreversion
 Source: "..\applications\tari_merge_mining_proxy\windows\runtime\start_xmrig.bat"; DestDir: "{app}\runtime"; Flags: ignoreversion
-Source: "..\common\config\presets\tari_config_example.toml"; DestDir: "{app}\config"; DestName: "config.toml"; Flags: ignoreversion
+; Note: please run generate_config.bat to generate this file first
```
Comment (hansieodendaal) [inline buildtools/windows_inno_installer.iss]: Hi @delta1, please add in a `[PreCompile] Section` into `windows_inno_installer.iss` to run the `generate_config.bat` automatically.

```
[PreCompile] Section

The [PreCompile] section is optional, and specifies any number of programs to execute before the Inno Setup 
compiler has started it's compilation.

The following is an example of a [PreCompile] section

[PreCompile]
Name: "Pre-Compile-Build.bat"; Parameters: ""/Build v1.3.33""; Flags: abortonerror cmdprompt
```

**Review by hansieodendaal — COMMENTED** [review]
[_I only reviewed changes to the Inno Setup Script_] 

Please see the comment below. Thanks.

File: buildtools/windows_inno_installer.iss (hunk 3 above)
Comment (delta1) [reply buildtools/windows_inno_installer.iss]: I did try that previously, but it didn't work because I had it after [files]. Anyway, done now.

**Review by hansieodendaal — COMMENTED** [review]
Checked it out, but I also could not get this to work

**Comment by delta1** [comment]
I did test it on windows working fine

**Review by stringhandler — DISMISSED** [review]

**Comment by delta1** [comment]
had a random integration test failure after the most recent push, just checking it again

**Review by stringhandler — APPROVED** [review]
Really keen for this, thanks

**Pull request merged**